src/lambdaconnect_model/utils.cljc

```diff
@@ -1,8 +1,38 @@
 (ns lambdaconnect-model.utils
-  (:require [clojure.set :refer [union]]))
+  (:require 
+   #?(:clj [clojure.spec.alpha :as s] 
+      :cljs [cljs.spec.alpha :as s])
+   [clojure.set :refer [union]]))
 
-(defmacro functionise [macro]
-  `(fn [& args#] (eval (cons '~macro args#))))
+(def pmap #?(:clj pmap :cljs map))
+
+#?(:clj 
+   (defmacro defspec [k spec-form]
+     `(s/def-impl ~k (quote ~spec-form) ~spec-form)))
+
+(defn fmap [f m] 
```

Comment:
`clojure.algo.generic/fmap` zachowuje każdą strukturę, a to jest funkcja z mapy w mapę, lepiej nazwać to `map-keys`, żeby nikt tego przypadkiem nie użył na liście/secie...